src/modules/fundingManager/bondingCurveFundingManager/BondingCurveFundingManagerBase.sol

```diff
@@ -253,6 +265,67 @@ abstract contract BondingCurveFundingManagerBase is
         emit TokenDecimalsUpdated(oldDecimals, tokenDecimals);
     }
 
+    /// @dev Returns the collateral and issuance fee percentage retrieved from the tax manager for
+    ///     buy operations
+    /// @return collateralBuyFeePercentage The percentage fee to be collected from the collateral
+    ///     token being deposited for minting issuance, expressed in BPS
+    /// @return issuanceBuyFeePercentage The percentage fee to be collected from the issuance token
+    ///     being minted, expressed in BPS
+    function _getProtocolBuyFee()
+        internal
+        virtual
+        returns (uint collateralBuyFeePercentage, uint issuanceBuyFeePercentage)
+    {
+        // TODO: Add actual function call
+        // return __Module_orchestrator.taxMan().getProtocolBuyFee();
+        return (100, 100);
+    }
+
+    function _collectFeeViaTransfer(
+        IERC20 _token,
+        uint _transactionAmount,
+        uint _feePct
+    ) internal returns (uint netAmount) {
+        // skip protocol fee collection if fee percentage set to zero
+        if (_feePct == 0) return _transactionAmount;
+
+        // get net and fee amount
+        uint feeAmount;
+        (netAmount, feeAmount) =
+            _calculateNetAmountAndFee(_transactionAmount, _feePct);
+        // transfer fee amount
+
+        // TODO: Get address of where the fee needs to be sent
+        // _token.safeTransferFrom(
+        //     address(__Module_orchestrator.taxMan()), feeAmount
+        // );
+
+        // TODO: Should we add event to protocol fee collection?
+
+        return netAmount;
+    }
+
+    function _collectFeeViaMinting(uint _transactionAmount, uint _feePct)
+        internal
+        returns (uint netAmount)
+    {
+        // skip protocol fee collection if fee percentage set to zero
+        if (_feePct == 0) return _transactionAmount;
+
+        uint feeAmount;
+        // get net and fee amount
+        (netAmount, feeAmount) =
+            _calculateNetAmountAndFee(_transactionAmount, _feePct);
+
+        // mint fee amount
+        // TODO: Get address of where the tokens need to be minted to
+        // _mint(address(__Module_orchestrator.taxMan()), feeAmount);
+
+        // TODO: Should we add event to protocol fee collection?
```

Comment:
Good question
If we do this then maybe even when the protocol takes their share
Seems kind of overkill, but on the other side makes it easy for users to see what amounts of tax have been taken

Reply:
I'd be in favor. Just to be able to easily reconstruct the process. We'll probably be interested in tracking all the amounts sent to the TaxMan and why